plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/internal/N4JSEclipseModel.java

```diff
@@ -80,15 +80,15 @@ public N4JSEclipseProject getN4JSProject(URI location) {
 		}
 
 		final String projectName = location.lastSegment();
-		final IProject project;
-		if (location.isFile()) {
+		IProject project;
+		project = workspace.getProject(projectName);
+		if (project == null && location.isFile()) {
 			project = externalLibraryWorkspace.getProject(projectName);
 			checkNotNull(project, "Project does not exist in external workspace. URI: " + location);
```

Comment:
this check is obsolete now

Reply:
That checked caused the "Run As" to not show the possible runners.